Remove superfluous checks - NonNull::new already checks pointer for null

Mingun opened the pull request:
They was introduced in the first god commit 8e6866f43a3f6b4de782b44c5c0d72a15994f63c

EvinceMoi commented:
I just want to parse a simple yaml file, but why should I need to compile with tokio(come with rlg crate), it makes completely no sense.

sebastienrousseau commented:
@EvinceMoi Fair point and thanks for raising it. Changes in the next release and PR should fix this. https://github.com/sebastienrousseau/serde_yml/pull/24

sebastienrousseau commented:
@asomers and @evaneaston Changes merged to v0.0.7. Thanks!

Mingun committed:
Remove superfluous checks - NonNull::new already checks pointer for null

They was introduced in the first god commit 8e6866f43a3f6b4de782b44c5c0d72a15994f63c
